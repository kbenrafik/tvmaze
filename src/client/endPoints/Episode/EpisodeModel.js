import Model from '../../Model';

/**
 * Episode model
 * @class
 */
class EpisodeModel extends Model {
  /**
   * Get Season
   * @returns {String}
   */
  getSeason() {
    return this.get('season');
  }
}

export default EpisodeModel;