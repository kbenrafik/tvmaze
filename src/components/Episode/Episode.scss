.episode {
  &__wrapper {
    display: flex;
    margin-bottom: 20px;
  }

  &__cover {
    margin-right: 15px;
    img {
      width: 200px;
    }
  }

  &__name {
    font-weight: bold;
    margin-bottom: 10px;
  }

  &__summary {

  }
}