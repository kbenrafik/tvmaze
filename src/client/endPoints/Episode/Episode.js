import Http from '../../Http';
import { GET } from '../../constants';
import { routes } from '../../routes';
import EpisodeModel from './EpisodeModel';

/**
 * Season endPoint
 * @class
 *
 * @property {SeasonModel[]} listByShow : Fetches a list of episodes by show ID on the tvmaze.
 * @property {SeasonModel[]} listBySeason : Fetches a list of episodes by season ID on the tvmaze.
 * @property {SeasonModel[]} fetch : Fetches a single episode by ID on the tvmaze.
 */
class Episode extends Http {
  /**
   * Fetches a list of episodes by show ID on the tvmaze.
   *
   * @example
   * const episodes = await client.episode.listByShow(idShow);
   * // Do something with the episodes
   *
   * @param {String} idShow to fetch
   * @returns {EpisodeModel[]}
   */
  async listByShow(idShow) {
    const episodes = await this.response(GET, routes.episodesByShowPath(idShow));
    return episodes
      .map(episode => new EpisodeModel(episode));
  }

  /**
   * Fetches a list of episodes by season ID on the tvmaze.
   *
   * @example
   * const episodes = await client.episode.listBySeason(idSeason);
   * // Do something with the episodes
   *
   * @param {String} idSeason to fetch
   * @returns {EpisodeModel[]}
   */
  async listBySeason(idSeason) {
    const episodes = await this.response(GET, routes.EpisodesBySeasonPath(idSeason));
    return episodes
      .map(episode => new EpisodeModel(episode));
  }

  /**
   * Fetches a single episode by ID on the tvmaze.
   *
   * @example
   * const episode = await client.episode.fetch(idEpisode);
   * // Do something with the episode
   *
   * @param {String} is of the episode to fetch.
   * @returns {EpisodeModel}
   */
  async fetch(idEpisode) {
    const episode = await this.response(GET, routes.episodePath(idEpisode));
    return new EpisodeModel(episode);
  }
}

export default Episode;