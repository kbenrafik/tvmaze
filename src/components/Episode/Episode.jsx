import React from 'react';
import {
  Link
} from 'react-router-dom';
import './Episode.scss';

/**
 * Component for single episode
 * @param {EpisodeModel} episode
 * @param {String} idShow
 * @returns {null|*}
 * @constructor
 */
const Episode = ({
  episode,
  idShow
}) => {
  if (!episode) {
    return null;
  }

  return (
    <div className="episode__wrapper">
      <div className="episode__cover">
        <Link to={`/show/${idShow}/episode/${episode.getId()}`}>
          <img src={episode.getImage('medium')} alt={episode.getName()}/>
        </Link>
      </div>
      <div className="episode__detail">
        <h3 className="episode__name">
          {episode.getName()}
        </h3>
        <div className="episode__summary" dangerouslySetInnerHTML={{__html: episode.getSummary()}}/>
      </div>
    </div>
  );
};

export default Episode;