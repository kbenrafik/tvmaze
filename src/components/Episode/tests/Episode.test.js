import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Episode from '../';
import EpisodeModel from '../../../client/endPoints/Episode/EpisodeModel';

test('should return null', () => {
  const tree = renderer
    .create(<Episode/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('should render episode', () => {
  const mockEpisode = new EpisodeModel({
    id: 1,
    summary: 'this is summary',
    name: 'episode name',
    image: {
      medium: '/link/image'
    }
  });
  const tree = renderer
    .create(<Router>
      <Episode episode={mockEpisode} idShow="659"/>
    </Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});