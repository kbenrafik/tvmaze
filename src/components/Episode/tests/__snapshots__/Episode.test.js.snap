// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render episode 1`] = `
<div
  className="episode__wrapper"
>
  <div
    className="episode__cover"
  >
    <a
      href="/show/659/episode/1"
      onClick={[Function]}
    >
      <img
        alt="episode name"
      />
    </a>
  </div>
  <div
    className="episode__detail"
  >
    <h3
      className="episode__name"
    >
      episode name
    </h3>
    <div
      className="episode__summary"
      dangerouslySetInnerHTML={
        Object {
          "__html": "this is summary",
        }
      }
    />
  </div>
</div>
`;

exports[`should return null 1`] = `null`;
